test/unit_test/character_sheet_unit_tests/FocusesSetUnitTests.java

```diff
@@ -28,18 +29,24 @@ public void expectException_SetFocusesMalformedInput() {
 		characterSheet.setData(Fields.FOCUSES, malformedInput);
 	}
 
-	// functional tests
 	@Test
-	public void testSetFocusWithFocusesSet() {
-		CharacterSheet characterSheet = new CharacterSheet("TestCharacterSheet");
-		FocusesSet expectedFocuses = new FocusesSet(new Focus(Focuses.ETIQUETTE));
-		characterSheet.setData(Fields.FOCUSES, expectedFocuses);
-		FocusesSet actualFocus = characterSheet.getData(Fields.FOCUSES);
-		assertEquals(expectedFocuses, actualFocus);
+	public void expectException_SetSameFocusesMoreThanAllowed() {
+		expectExceptionWithMessage(InvalidFocusesSetException.class, "ANIMAL_HANDLING is already improved!");
+		FocusesSet focusesSet = new FocusesSet(Focuses.ANIMAL_HANDLING, Focuses.ANIMAL_HANDLING);
+		focusesSet.add(Focuses.ANIMAL_HANDLING);
 	}
 
 	@Test
-	public void testSetFocusWithFocusesEnum() {
+	public void expectException_GetFocusesWithWrongParameter() {
+		expectExceptionWithMessage(InvalidFocusesSetException.class, "BRAWLING is not in the list!");
+		FocusesSet focusesSet = new FocusesSet();
+		focusesSet.getFocus(Focuses.BRAWLING);

```

Comment:
Nem arról volt szó, hogy a getFocus-tól megszabadulunk, és cserébe az iterátorral kérjük ki a megfelelő (egyetlen) tesztelendő elemet? Miért indokolt még a getFocus használata? ez egyébként a tesztet is feleslegessé teheti, hiszen most magát a getFocus metódust teszteled, de ha nincs rá szükség akkor tesztelni sem kell ugye.

Reply:
megvan.